h2o-r/h2o-package/R/models.R

```diff
@@ -471,11 +481,7 @@ predict.H2OModel <- function(object, newdata, ...) {
 #' @rdname predict.H2OModel
 #' @export
 h2o.predict <- function(object, newdata, ...){
-  if(class(object) == "H2OAutoML"){
```

Comment:
@seb-h2o it looks like it could be implemented via generic function (for `predict`) and dispatch concept. I'm just curious how `UseMethod` is working. It looks like it is built in function... but it is strange that name is camel case :)

Reply:
it's doing exactly what what you said: it defines "predict" as a generic function, and the dispatching is done based on the object type.
e.g. if object is a H2OAutoML, it dispatches to predict.H2OAutoML

Reply:
@seb-h2o great! good to know!